### PR TITLE
Pydantic v2 and FastAPI as core dependency

### PR DESCRIPTION
**Pull request by pierrejeambrun**
As discussed here https://github.com/apache/airflow/pull/41798, pydantic v2 will be part of core requirement now. Mostly because fastapi requires pydantic. Even if fastapi can still (for now) work with pydantic v1, airflow 3 will only support pydantic v2.

### CODE REVIEW

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
The failing `Test Open API` have been fixed by https://github.com/apache/airflow/pull/41856#issuecomment-2317618682

**Comment by pierrejeambrun** [comment]
Thanks @potiuk for the tips, rebased.

**Comment by pierrejeambrun** [comment]
🤔 the db tests are still failing due to the unique constraint on dag run failing. Not sure how this is related to pydantic

**Comment by potiuk** [comment]
Screenshot 2024-08-29 at 16 14 38

File: tests/operators/test_python.py [hunk 1]
```diff
@@ -1085,15 +1081,7 @@ def f():
...
-        def f():
-            from airflow.operators.python import get_current_context
-
-            get_current_context()
-            return []
-
-        error_msg = "`get_current_context()` needs to be used with Pydantic 2 and AIP-44 enabled."
+        error_msg = "`get_current_context()` needs to be used with AIP-44 enabled."
         with pytest.raises(AirflowException, match=re.escape(error_msg)):
             self.run_as_task(f, return_ti=True, multiple_outputs=False, use_airflow_context=True)
```
Comment (potiuk) [inline tests/operators/test_python.py:1087]:
```suggestion
```

Comment (potiuk) [inline tests/operators/test_python.py]:
```suggestion

```

Comment (pierrejeambrun) [reply tests/operators/test_python.py]: Ohh well, that explains a lot 😢. Thanks for spotting that.

Comment (potiuk) [reply tests/operators/test_python.py]: missed it too when reviewing 🤷

**Review by jedcunningham — APPROVED** [review]

**Pull request merged**